src/index.tsx

```diff
@@ -498,147 +588,193 @@ class DiffViewer extends React.Component<
+          }
+          if (
+            line.left.type === DiffType.DEFAULT
+						&& (currentPosition > extraLines
+							|| typeof diffBlockStart === 'undefined')
+						&& !this.state.expandedBlocks.includes(diffBlockStart)
+          ) {
+            skippedLines.push(i + 1);
+
+            if (i === lineInformation.length - 1 && skippedLines.length > 1) {
+              const skippedLinesIndicator = this.renderSkippedLineIndicator(
+                skippedLines.length,
+                diffBlockStart,
+                line.left.lineNumber,
+                line.right.lineNumber,
+                isSkippedCode
+              );
+              isSkippedCode = true;
+              return skippedLinesIndicator;
+            }
+            return null;
+          }
+        }
+
+        const diffNodes = viewRenderer(line, i);
+
+        if (currentPosition === extraLines && skippedLines.length > 0) {
+          const { length } = skippedLines;
+          skippedLines = [];
+          const skippedLinesIndicator = this.renderSkippedLineIndicator(
+            length,
+            diffBlockStart,
+            line.left.lineNumber,
+            line.right.lineNumber,
+            isSkippedCode
+          );
+          isSkippedCode = true;
+          return (
+            <React.Fragment key={i}>
+              {skippedLinesIndicator}
+              {diffNodes}
+            </React.Fragment>
+          );
+        }
+        return diffNodes;
+      },
+    );
+  };
+
+  public render = (): JSX.Element => {
+    const {
+      oldValue,
+      newValue,
+      useDarkTheme,
+      leftTitle,
+      rightTitle,
+      splitView,
+	    verticalView,
+      hideLineNumbers,
+    } = this.props;
+
+    if (typeof oldValue !== 'string' || typeof newValue !== 'string') {
+      throw Error('"oldValue" and "newValue" should be strings');
+    }
+
+    this.styles = this.computeStyles(this.props.styles, useDarkTheme);
+    const colSpanOnSplitView = hideLineNumbers ? 2 : 3;
+    const colSpanOnInlineView = hideLineNumbers ? 2 : 4;
+
+    if (splitView && verticalView) {
+      const nodes = { 
+        left: this.renderDiff(this.renderLeftView), 
+        right: this.renderDiff(this.renderRightView) 
+      };
+      const leftTitleRow = (
+        <tr>
+          <td
+				  colSpan={colSpanOnInlineView}
+				  className={this.styles.titleBlock}>
+				  <pre className={this.styles.contentText}>{leftTitle}</pre>
+          </td>
+			  </tr>
+      );
+      const rightTitleRow = (
+        <tr>
+          <td
+            colSpan={colSpanOnInlineView}
+            className={this.styles.titleBlock}>
+            <pre className={this.styles.contentText}>{rightTitle}</pre>
+          </td>
+        </tr>
+      );
+
+	  return (
+        <div style={{ display: 'flex', flexDirection: 'column', gap: this.props.gap }}>
+          <table
+            className={cn(this.styles.diffContainer, {
+              [this.styles.splitView]: splitView && !verticalView,
```

Comment:
how can this work? if verticalView=false it won't be inside the block, unless I'm missing something?